### ISSUE TITLE
first glance at the games functionality

### ISSUE THREAD

**Issue by monzug** [alpheios-project/inflection-games#3]
1) design: can we add more space between Lexemes data and Games variants in the first pop-up? it looks a little crowded.
2) the Hide button next to Games variants does not seem to work.
3) design: would it be possible to make more clear what to click to start the game in Games variants section (first pop-up)?
4) amens is an adj of the 3rd decl. However, declension is not an option to select in the games. why?
5) it seems that when there are several lemmas for a word, the yellow highlighting goes to the one that has no meaning or no morphology or less common. e.g. deos (yellow is deus with no definition found) or magis (yellow is adverb, not the noun) or me (see attached screenshot)
6) going back to deos, in whitaker is 2nd declension
 de.os                N      2 1 ACC P M                 
Deus, Dei  N  M   [XEXAO]  
God (Christian text); god; divine essence/being, supreme being; statue of god;
in the game is 4th declension. where is this coming from?
7) design: can we suppress the Games button for pronoun, conjunction etc?
8) at one point, I had: made clicks 7, max clicks 6.... etc - what is the functionality of max clicks, then?
9) design: can we add a reset button to stop counting success, failed etc?
10) Cynthia is not highlighted in yellow in the morphology pop-up.
11) design: I would suggest to have a game with two options: 1) same as what we have now, 2) remove the declension, gender etc from the morphology pop-up so the user can really test his/her knowledge. Out of scope, right?
12) not sure why the yellow highlighted is sticky after clicking on a word. is this by design?
see screenshot
![screenshot_2018-10-11 alpheios sextus propertius elegies 1](https://user-images.githubusercontent.com/41396793/46811491-e1852300-cd40-11e8-895d-d973996dd4ce.png)

 also the show/hide when playing the games doesn't work properly

**Comment by balmas** [comment]
Re # 11 , nothing really is out of scope right now ,the whole point of this exercise is to figure out what might be in scope :-) 

Exactly how the use of the games should or shouldn't interact with the rest of the interface is up for discussion!

**Comment by monzug** [comment]
an other thing that I have notice: when I close the game pop-up and move the focus back to the morphology pop-up, the backscreen slides up few rows.
tested on firefox

**Comment by irina060981** [comment]
Hello, @monzug and @balmas!

Monica, thank you for finding time to look at the games!
My answers:

1) *design: can we add more space between Lexemes data and Games variants in the first pop-up? it looks a little crowded* 
I have tried to make as near as possible - to stay more space for the table.
But If you think - it is too near - I will add space.

2) *the Hide button next to Games variants does not seem to work.*
    there is a condtion there - that it could be hidden only after you have selected one.
    I will hide `hide` button if no game is selected - if it seems as it doesn't work

3) *design: would it be possible to make more clear what to click to start the game in Games variants section (first pop-up)?* 
   Ok, I will add - select a game instead of Game Variants

4) *amens is an adj of the 3rd decl. However, declension is not an option to select in the games. why?*
What do you mean here?
It the first in the Features List:
![image](https://user-images.githubusercontent.com/12377640/46866314-34a3c680-ce64-11e8-9ce4-72da5a75da61.png)
How could I reproduce the absence of declension?

5) Yellow highliting - it is for alignment property and diambiguated lemmas - not for games

6) *going back to deos, in whitaker is 2nd declension. in the game is 4th declension. where is this coming from?*
I don't see 4th declension here - could say me steps to reproduce?
![image](https://user-images.githubusercontent.com/12377640/46866425-b398ff00-ce64-11e8-91d8-8ed90fca053e.png)

7) *design: can we suppress the Games button for pronoun, conjunction etc?*
Yes I will fix it

8) *at one point, I had: made clicks 7, max clicks 6.... etc - what is the functionality of max clicks, then?*
From my point of view it means that you could make 6 clicks and the 7th will be the last ... if it is confusing - I will fix it - and the 6th will be the last (I don't remember the steps I was deciding how to handle with clicks now, but your question is saying - that my decision is not natural :) )

9) *design: can we add a reset button to stop counting success, failed etc?*
It is an interesting suggestion - I will add, thanks, Monica!

10) Yellow highliting - it is for alignment property and diambiguated lemmas - not for games

11) *design: I would suggest to have a game with two options: 1) same as what we have now, 2) remove the declension, gender etc from the morphology pop-up so the user can really test his/her knowledge. Out of scope, right?*
Now, it is not difficult to add such an option really, I will add it - need to think only how to organize the UI for this (it is so less space .... :) )

12) sticking and scrolling up - it is for lignment functionality - browser tries to show latin and aligning translation - but it is not for games

Ok, I have pointed what I could make on the next week and also I left some questions in the list, hope, you will give me some more explanations on it :)

**Comment by monzug** [comment]
Irina,
thanks for the reply. are we using the same build? I was able to reproduce
the missing declension for amens
see screenshots

**Comment by monzug** [comment]
here is the screenshot for deos of 4th decl

**Comment by irina060981** [comment]
![image](https://user-images.githubusercontent.com/41396793/46873924-e5c74400-ce05-11e8-8e10-eba0fc9e6057.jpg)

**Comment by irina060981** [comment]
![image](https://user-images.githubusercontent.com/41396793/46873924-e5c74400-ce05-11e8-8e10-eba0fc9e6057.JPG)

**Comment by irina060981** [comment]
Hello, Monica (@monzug )!
I am using 2 resources 
1) http://www-test.alpheios.net/alpheios-texts/Perseus.text.1999.02.0066/book1_poem1.html
    This resouce as with build for testing was given to me by @balmas 

**amens**
![image](https://user-images.githubusercontent.com/12377640/46934738-36f06580-d09c-11e8-91a8-df3a9b63605d.png)

**deos**
![image](https://user-images.githubusercontent.com/12377640/46934805-78811080-d09c-11e8-93c0-094a4374bf55.png)

2) http://alpheios-test4.irina-sklyarova.ru/Perseus.text.1999.02.0066/book1_poem2.html
    This is a testing copy on my server - where I am testing my changes
    The both resources has the same results (as on screenshots)

**Comment by irina060981** [comment]
Monica, it is really strange!
Because I couldn't imagine how it could be realized these differences
if declension is missing for amens in your case - it seems that the inflection game table differs too.

If the game table differs for amens and deos than it could mean that inflection tables differs on different builds, because I am coping it to games or there are some wonders here ...

Could you check in you build - if inflection tables are correct for these words?

**Comment by monzug** [comment]
in the case of deos, the scenario is different. we have two definition for
deos and the game is set for the less common word which does not have the
declension. I have compared the result with Whitaker and the second
definition (the less common one) does not exist. in the inflection table,
all the matching ending for deos (2nd decl nom and acc and 4th decl acc)
are highlighted. could it be that declension information for deos is coming
from matching the endings???? two things here:
1) the less common word is selected
2) based on the partial information on this word, the declension is chosen
by the matching

**Comment by monzug** [comment]
ok. I figured out what's going on here. I do click on amens and then from
the morphology pop-up, I click on the Games button and the scenario is the
one from screenshot above with no declension in the game.
if I click on Games from the lookup for amens (with no diagram involved)
the scenario is the one that Irina has reported (with the declension in the
game).
still in both games, I am not able to fail a game, they all end up  with
success at the 7th click

**Comment by irina060981** [comment]
Monica, thank you for all this investigation!

I was all in adding changes to the code from the above list (I have added them - 1, 2, 3, 7, 8, 9, 11).
now the window for games looks like this:
**(not hard mode)**
![image](https://user-images.githubusercontent.com/12377640/46953906-79cd3000-d0d2-11e8-80c3-8e4366a57b96.png)

**(hard mode)**
![image](https://user-images.githubusercontent.com/12377640/46953876-64580600-d0d2-11e8-812c-001254bf8960.png)

PR here:
https://github.com/alpheios-project/inflection-games/pull/6

Changes here:
https://github.com/alpheios-project/components/tree/infl-games

Rebuild and recompile here:
https://github.com/alpheios-project/embed-lib/tree/inflection-games

**Comment by irina060981** [comment]
@monzug, great, now I know how to try reproduce it, thanks!

About amens and deos - I haven't been able to find it in the text - so I was only looked it up.
So I will find it in the text and test.

About always successful - if to be honest if to click in the table - I am always failing :)
If to click on features - than it is really easy to win :)

Ok - I will work on such strange bahaviour with disambiguated words and also I have found several bugs more.
Again, Monica, thank you for your attentiveness!

**Comment by irina060981** [comment]
@balmas, thank you for merging embedded for inflection-games - I have failed to merge it quickly myself and then found out - that you had already done it!

**Comment by balmas** [comment]
> ok. I figured out what's going on here. I do click on amens and then from the morphology pop-up, I click on the Games button and the scenario is the one from screenshot above with no declension in the game. if I click on Games from the lookup for amens (with no diagram involved) the scenario is the one that Irina has reported (with the declension in the game). 

Ah ok, this is interesting. It means that something is wrong with the disambiguation. Declension is not provided in the treebank data, but the disamibugation is supposed to merge that data with the parser results without losing anything. But it would seem that is not the case.  It needs further investigation to see why, I've entered this as alpheios-project/components#249

**Comment by monzug** [comment]
Bridget,
there is also the deos issue that is slightly different (morphology from
lookup is different from morphology with diagram)

**Comment by balmas** [comment]
yep, will look at that in a minute.

**Comment by balmas** [comment]
I'm not sure if I am understanding everything about `deos` . I think it's related to alpheios-project/components#249 but a slightly different problem. 

The disambiguation is failing to find the right entry from the morphological output for `deos` I think because of the capitalization.  It should probably be doing a case-insensitive match, although I am not positive about this, because in fact capitalization can mean a different lemma, as in the case of a proper-name. (I.e God and god are quite legitimately different lemmas). (There is also an interesting parser behavior here with the variation of o vs u in the lemma, because deos gives you only the Deus, Dei lemma, whereas deus gives you both).

I think this in part confirms what I was expecting, that it will be difficult to have really solid games on text whose annotations haven't been fully curated and reviewed to make sure we have the full correct data for it. 

The question of which declension is used - it does seem if it is pulling that from the matching cell in the table. I'm not sure that's actually wrong behavior?

**Comment by monzug** [comment]
no, it's not except that deus does not exist in Whitaker. it does create
the entry ad hoc, that's my impression.

**Comment by balmas** [comment]
yes that's correct. It's not really ad-hoc, it's creating the entry from the treebank data, which specifies the lemma as 'deus' rather than 'Deus'. In this case, I think actually the treebank data is wrong, the lemma should be Deus.

**Comment by monzug** [comment]
actually it should be 'deos'

**Comment by monzug** [comment]
and 'me' is same scenario as deos.

**Comment by irina060981** [comment]
As for `amens` - there is a problem with filtering inflections for disambiguated lexemes (it couldn't filter only `masculine` for example from `masculine femine neuter`

The whole description is published here
https://github.com/alpheios-project/components/issues/249#issuecomment-430267328

**Comment by monzug** [comment]
an other one similar to deos is fugiendo. it does look for fugio not
fugiendo. the matching in the inflection table is wrong.

**Comment by monzug** [comment]
Most issues are fixed or have been moved to new issues. 
the only three that haven't been address are the following:
- Cynthia is not highlighted in yellow in the morphology pop-up.
- design: I would suggest to have a game with two options: 1) same as what we have now, 2) remove the declension, gender etc from the morphology pop-up so the user can really test his/her knowledge. Out of scope, right?
- not sure why the yellow highlighted is sticky after clicking on a word. is this by design?
this is still happening in latest build. every time I click on a word it gets highlighted and then it stays till I click on it again. why?

One more for @irina060981 - why there is (hard mode) in the Game header next to the word?
Thanks

**Comment by irina060981** [comment]
Hello, Monica :)

*Hard mode* - is the game mode without features list :)
Click on it and the game would be able to be won only by clicking on the table .

So there is an answer for the second point and the last question at a time!

About yellow highlighting - it concerns the alignment ability (it is not inside the inflection games).
But anyway I could answer your last questions about it:
Yes it is by design - the idea of yellow highlighting is to show at the same time the word and its translation in the texts. As the translation and the text could be placed far from each other it was suggested to add some additional behaviour.
So if you click on a word and it has aligned translated they both get highlighting and it sticks to them, because at the second step the screen could be scrolled to the translation or vice versa if one of them is out of the screen.
This way we "mark" translations pairs. If you click the second time it will loose the highlighting.

**Comment by monzug** [comment]
Irina, Thanks you for prompt reply.
as per hard mode, is it a temporary addition? if not, I do not think it is
easy to understand what is suppose to do.
